### PR TITLE
Tree-Shake OperationSource

### PR DESCRIPTION
**Pull request by schmidt-sebastian**

### CODE REVIEW

File: packages/database/src/core/operation/Operation.ts [hunk 1]
```diff
@@ -42,40 +41,38 @@ export interface Operation {
...
 
-  static user() {
-    return new OperationSource(
-      /*fromUser=*/ true,
-      false,
-      null,
-      /*tagged=*/ false
-    );
-  }
+export function newOperationSourceUser() {
```
Comment (Feiyang1) [inline packages/database/src/core/operation/Operation.ts:51]:
```suggestion
export function newOperationSourceUser(): OperationSource {
```

File: packages/database/src/core/operation/Operation.ts [hunk 2]
```diff
@@ -42,40 +41,38 @@ export interface Operation {
...
 
-  static server() {
-    return new OperationSource(
-      false,
-      /*fromServer=*/ true,
-      null,
-      /*tagged=*/ false
-    );
-  }
+export function newOperationSourceServer() {
```
Comment (Feiyang1) [inline packages/database/src/core/operation/Operation.ts:60]:
```suggestion
export function newOperationSourceServer(): OperationSource {
```

**Review by Feiyang1 — CHANGES_REQUESTED** [review]
Perhaps we can start turning on eslint rules that we turned off for database as part of the rewrite?

**Comment by schmidt-sebastian** [comment]
> Perhaps we can start turning on eslint rules that we turned off for database as part of the rewrite?

Is it possible to do this only for modified files?

File: packages/database/src/core/operation/Operation.ts [hunk 3]
```diff
@@ -42,40 +41,38 @@ export interface Operation {
...
 
-  static user() {
-    return new OperationSource(
-      /*fromUser=*/ true,
-      false,
-      null,
-      /*tagged=*/ false
-    );
-  }
+export function newOperationSourceUser(): OperationSource {
```
Comment (hsubox76) [inline packages/database/src/core/operation/Operation.ts:51]: Why are these all prefixed with "new"?

Comment (schmidt-sebastian) [reply packages/database/src/core/operation/Operation.ts:51]: They are factory functions. I can remove the prefix if you think we shouldn't do this for consistency - let me know,

Comment (hsubox76) [reply packages/database/src/core/operation/Operation.ts:51]: That makes sense.

**Review by hsubox76 — APPROVED** [review]

**Pull request merged**